common/djangoapps/student/models_api.py

```diff
@@ -62,3 +66,76 @@ def get_phone_number(user_id):
         log.exception(exception)
         return None
     return student.phone_number or None
+
+
+def is_username_retired(username):
+    """
+    Checks to see if the given username has been previously retired
+    """
+    locally_hashed_usernames = user_util.get_all_retired_usernames(
+        username,
+        settings.RETIRED_USER_SALTS,
+        settings.RETIRED_USERNAME_FMT
+    )
+
+    # TODO: Revert to this after username capitalization issues detailed in
+    # PLAT-2276, PLAT-2277, PLAT-2278 are sorted out:
+    # return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
+
+    # Avoid circular import issues
+    from openedx.core.djangoapps.user_api.models import UserRetirementStatus
+
+    # Sandbox clean builds attempt to create users during migrations, before the database
+    # is stable so UserRetirementStatus may not exist yet. This workaround can also go
+    # when we are done with the username updates.
+    try:
+        return _User.objects.filter(username__in=list(locally_hashed_usernames)).exists() or \
+            UserRetirementStatus.objects.filter(original_username=username).exists()
+    except ProgrammingError as exc:
+        # Check the error message to make sure it's what we expect
+        if "user_api_userretirementstatus" in text_type(exc):
+            return User.objects.filter(username__in=list(locally_hashed_usernames)).exists()
+        raise
+
+
+def is_email_retired(email):
+    """
+    Checks to see if the given email has been previously retired
+    """
+    locally_hashed_emails = user_util.get_all_retired_emails(
+        email,
+        settings.RETIRED_USER_SALTS,
+        settings.RETIRED_EMAIL_FMT
+    )
+
+    return _User.objects.filter(email__in=list(locally_hashed_emails)).exists()
+
+def get_user_last_login_by_username(username):
```

Comment:
I used `last_login` as a proxy for the time a user requested deletion. But then I realized I could use the `modified` field on the `UserRetirementRequest` model. But I'm not sure whether there's a difference. I'm leaving this here until I can chat with Data Engineering about it.